udpa/annotations/security.proto

```diff
@@ -0,0 +1,21 @@
+syntax = "proto3";
+
+package udpa.annotations;
+
+import "google/protobuf/descriptor.proto";
+
+extend google.protobuf.FieldOptions {
+  // Magic number is the 28 most significant bits in the sha256sum of
+  // "udpa.annotations.security".
+  FieldSecurityAnnotation security = 11122993;
+}
+
+// These annotations indicate metadata for the purpose of understanding the
+// security significance of fields.
+message FieldSecurityAnnotation {
```

Comment:
qq: is this going to work for primitive fields? I'm not sure how often that is an issue but it might be?

Reply:
Don't see why it wouldn't, it's just an annotation. Tools can check that the right value is there eventually, even if it is the primitive default.

Reply:
But if this annotation is only annotating that something is "set" will that work for primitive fields? Should we also include some concept of a secure default? Maybe that gets too complicated for this purpose.

Reply:
I was thinking about using this to begin with to just mechanically generate a list of fields that should be set, e.g. in our edge configuration FAQ. The idea was to give folks instructions "make sure to consider setting this if you have an untrusted upstream" (even if the primitive default is that setting).

That said, I did have the same thought that specifying secure defaults would be a good start. What I propose then is 3 fields:
1. `bool configure_for_untrusted_downstream`
2. `bool configure_for_untrusted_upstream`
3. A nested PGV annotation giving suggested secure defaults. This is very flexible, as we get a fairly expressive language off-the-shelf.

Does that sound like it would make sense?

Reply:
Yes +1

Reply:
I played around with (3) a bit and didn't like the results since:
1. PGV doesn't allow you to specify message or repeated literals for == relationships, which is a big limitations, forcing you to work at primitive field level.
2. In the absence of CEL, we don't have the expressivity to connect fields.

I'm going to work on instead providing a single example fragment via a YAML or text proto string for (3)